File: 2021/07/06/main.go
package main

import "sort"

func minSetSize(arr []int) int {
	l, s := []int{}, make(map[int]int)
	for _, a := range arr {
		s[a]++
	}
	for _, v := range s {
		l = append(l, v)
	}
	sort.Slice(l, func(i, j int) bool {
		return l[i] > l[j]
	})
	c := 0
	for i, p := range l {
		c += p
		if c*2 >= len(arr) {
			return i + 1
		}
	}
	return 0
}

File: 2021/07/06/main.py
class Solution:
    def minSetSize(self, arr: List[int]) -> int:
        c, s, m = 0, {}, len(arr) / 2
        for a in arr:
            if a in s:
                s[a] += 1
            else:
                s[a] = 1
        s = sorted(s.items(), key=lambda x: -x[1])
        for i, p in enumerate(s):
            c += p[1]
            if c >= m:
                return i + 1